### PR TITLE
FEATURE: LX-156 Filter Attr. Elements by specific date filters

### PR DESCRIPTION
**Pull request by NestorEncinas**
JIRA: LX-156

---

> [!IMPORTANT]
> Please, **don't forget to run `rush change`** for the commits that introduce **new features** 🙏

---

Refer to [documentation](https://github.com/gooddata/gooddata-ui-sdk/blob/master/dev_docs/continuous_integration.md) to see how to run checks and tests in the pull request. This is the list of the most used commands:

```
extended test - backstop
```

```
// Tiger platform
extended test - tiger-cypress - integrated
extended test - tiger-cypress - isolated
extended test - tiger-cypress - record
```

```
// Bear platform
extended test - cypress - integrated
extended test - cypress - isolated
extended test - cypress - record
```

### CODE REVIEW

**Comment by gdgate** [comment]
Build failed (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/18273/ : SUCCESS in 3s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/19911/ : FAILURE in 5m 54s

**Comment by gdgate** [comment]
Build failed (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/18279/ : SUCCESS in 3s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/19916/ : FAILURE in 2m 34s

**Comment by gdgate** [comment]
Build failed (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/18286/ : SUCCESS in 3s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/19926/ : FAILURE in 2m 21s

**Comment by gdgate** [comment]
Build failed (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/18302/ : SUCCESS in 3s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/19944/ : FAILURE in 4m 16s

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/18305/ : SUCCESS in 3s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/19948/ : SUCCESS in 11m 05s

**Comment by gdgate** [comment]
Build failed (check-extended pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/18308/ : SUCCESS in 3s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-pull-request-dispatcher-pipeline/2050/ : UNSTABLE in 33m 10s

**Comment by NestorEncinas** [comment]
extended test - cypress - integrated

**Comment by gdgate** [comment]
Build in pipeline check-extended aborted.

**Comment by NestorEncinas** [comment]
ok to test

**Comment by gdgate** [comment]
Build failed (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/18315/ : SUCCESS in 3s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/19957/ : FAILURE in 19s

**Comment by NestorEncinas** [comment]
ok to test

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/18321/ : SUCCESS in 3s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/19964/ : SUCCESS in 11m 16s

**Comment by NestorEncinas** [comment]
extended test - tiger-cypress - isolated

**Comment by gdgate** [comment]
Build succeeded (check-extended pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/18322/ : SUCCESS in 3s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-pull-request-dispatcher-pipeline/2052/ : SUCCESS in 12m 52s

**Comment by NestorEncinas** [comment]
extended test - cypress - isolated

**Comment by gdgate** [comment]
Build succeeded (check-extended pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/18324/ : SUCCESS in 3s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-pull-request-dispatcher-pipeline/2053/ : SUCCESS in 10m 47s

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/18341/ : SUCCESS in 3s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/19989/ : SUCCESS in 11m 13s

File: libs/sdk-ui-dashboard/src/presentation/filterBar/attributeFilter/useDependentDateFilters.ts [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+ *
+ * @beta
+ */
+export const useDependentDateFilters = (filter: IDashboardAttributeFilter): UseDependentDateFiltersResult => {
+    const allDateFilters = useDashboardSelector(selectFilterContextDateFiltersWithDimension);
+
+    const dependentDateFilters = useMemo(() => {
+        return filter.attributeFilter.filterElementsByDate?.map((parent) => {
+            const matchingFilter = allDateFilters.find(
+                (filter) => (filter.dateFilter.dataSet as any).identifier === parent.filterLocalIdentifier,
```
Comment (xMort) [inline libs/sdk-ui-dashboard/src/presentation/filterBar/attributeFilter/useDependentDateFilters.ts:29]: Does this needs to be `any`? Could not this by handled by objRefToString or some typeguard?

Comment (NestorEncinas) [reply libs/sdk-ui-dashboard/src/presentation/filterBar/attributeFilter/useDependentDateFilters.ts]: Removed any, missed it from todo list. Thanks.

File: libs/sdk-ui-dashboard/src/presentation/filterBar/attributeFilter/dashboardDropdownBody/configuration/hooks/useDependentDateFiltersConfiguration.ts [hunk 2]
```diff
@@ -1,34 +1,98 @@
...
+            const dateFilters: IDashboardAttributeFilterByDate[] = [];
+            dependentDateFilters.forEach((dependentDateFilter) => {
+                if (!dependentDateFilter.isSelected) {
+                    return;
+                }
+
+                dateFilters.push({
+                    filterLocalIdentifier: dependentDateFilter.localIdentifier,
+                });
+            });
```
Comment (xMort) [inline libs/sdk-ui-dashboard/src/presentation/filterBar/attributeFilter/dashboardDropdownBody/configuration/hooks/useDependentDateFiltersConfiguration.ts:81]: Highly optional but this is 3 lines vs 9 and is bit more readable (at least to me):

```
const dateFilters = dependentDateFilters
   .filter((filter) => filter.isSelected())
   .map((filter) => ({  filterLocalIdentifier: filter.localIdentifier }));
```

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/18371/ : SUCCESS in 3s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/20008/ : SUCCESS in 11m 35s

**Review by xMort — APPROVED** [review]

**Comment by yenkins** [comment]
### Sonar scan result


More detail, see in https://sonarqube-gate.intgdc.com/dashboard?id=gooddata-ui-sdk-gate-PR4774

To scan for vulnerabilities in dependencies and run unit tests (to get coverage report in sonar) please comment your PR with 'extended check sonar'.

**Comment by gdgate** [comment]
Build succeeded (gate pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-sonarqube-scan-zuul-gate/4073/ : SUCCESS in 10m 01s
- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-fossa-scanning-zuul/1952/ : SUCCESS in 1m 14s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/20009/ : SUCCESS in 11m 23s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-cypress-tiger-isolated-zuul-pipeline/1743/ : SUCCESS in 12m 08s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-cypress-bear-isolated-zuul-pipeline/1730/ : SUCCESS in 10m 17s
- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-merge-compliance-check/5247/ : SUCCESS in 6s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-compatibility-pipeline/676/ : SUCCESS in 5m 03s

**Pull request merged**